client/utils/clone-array.js

```diff
@@ -0,0 +1,37 @@
+/**
```

Comment:
The `clone` [doc snippet](http://underscorejs.org/#clone) says it only makes a shallow copy, however we need a deep clone. Underscore itself doesn't offer a `deepClone` function, and [digging into alternatives](https://github.com/jashkenas/underscore/issues/162) leads into using a ~3 year old, test-less [npm package](https://github.com/mateusmaso/underscore.deepclone), or rolling our own (either using a couple of Underscore fn's or using the `JSON.parse(JSON.stringify(...))` business).

IMO I'd prefer to maintain our own util file instead of farming it out to another NPM package (seeing as Underscore doesn't offer what we need). And, fwiw, there are at least tests written already for our utils. What do you think, @rehandalal ?